services/movie_session.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    queryset = MovieSession.objects.all()
+
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date,)
+
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: datetime = None,
+                         movie_id: int = None,
+                         cinema_hall_id: int = None) -> None:
+
+    if show_time:
+        movie_session = MovieSession.objects.get(id=session_id)
+        movie_session.show_time = show_time
+        movie_session.save()
+
+    if movie_id:
+        movie_session = MovieSession.objects.get(id=session_id)
+        movie_session.movie_id = movie_id
+        movie_session.save()
+
+    if cinema_hall_id:
+        movie_session = MovieSession.objects.get(id=session_id)
+        movie_session.cinema_hall_id = cinema_hall_id
+        movie_session.save()
```

Comment:
why do u need to make 3 query to a db ? why don't u use variable ?

Reply:
I did something like that already, but I was rejected. I'll change it back, but with a current .save() functions.